### PR TITLE
feat: Inital Commit to start up backend repository

### PR DESCRIPTION
**Pull request by RafaDSan**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PORT=
+DEFAULT_API_PREFIX=
+NODE_ENV=
```
Comment (FrancoAguzzi) [inline .env.template]:
```suggestion
NODE_ENV=

```

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+dist/
+jest.config.ts
```
Comment (FrancoAguzzi) [inline .eslintignore]:
```suggestion
jest.config.ts

```

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Default owners for everything in the repo
+
+@RafaDSan
```
Comment (FrancoAguzzi) [inline CODEOWNERS]:
```suggestion
@RafaDSan

```

File: src/app.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { envs } from "./core/config/env";
+import { Server } from "./server";
+
+(() => {
+	main();
+})();
```
Comment (FrancoAguzzi) [inline src/app.ts]: It is good to call function after they were declared

Comment (lgahdl) [inline src/app.ts]: You should have the same result just calling main (after the function declaration);
```suggestion
main();
```

File: src/server.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	constructor(options: ServerOptions) {
+		const { port } = options;
+		this.port = port;
+	}
+
+	async start(): Promise<void> {
+		//* Middlewares
+		this.app.use(express.json()); // parse json in request body (allow raw)
+		this.app.use(express.urlencoded({ extended: true })); // allow x-www-form-urlencoded
+		this.app.use(compression());
```
Comment (lgahdl) [inline src/server.ts:26]:
```suggestion
		this.app.use(compression());
		this.app.use(cors());
```

**Review by lgahdl — CHANGES_REQUESTED** [review]

File: src/app.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+function main(): void {
+	const server = new Server({
+		port: envs.PORT,
+		apiPrefix: "/api"
+	});
+	void server.start();
```
Comment (lgahdl) [inline src/app.ts]:
```suggestion
	await server.start();
```

File: src/app.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { envs } from "./core/config/env";
+import { Server } from "./server";
+
+(() => {
+	main();
+})();
+
+function main(): void {
```
Comment (lgahdl) [inline src/app.ts]:
```suggestion
async function main(): Promise<void> {
```

**Review by lgahdl — CHANGES_REQUESTED** [review]

**Pull request merged**